Add the `get_locator_from_height` function

stanislav-tkach opened the pull request.

stanislav-tkach commented:
I have found only a minor improvement: to send the locator starting from the block preceding the tip, so the `best_known_block` field is initialized even for the peers that are fully synced with our node.

TheQuantumPhysicist commented on p2p/src/sync/peer.rs:
```diff
@@ -277,11 +283,21 @@ where
+        if headers.len() == 1 {
+            let block_id = headers.first().expect("Headers shouldn't be empty").block_id();
+            if let Some(index) =
+                self.chainstate_handle.call(move |c| c.get_block_index(&block_id)).await??
```
Nitpick: I would put this in its own line

stanislav-tkach commented:
After a discussion I have reverted the changes in the sync manager and updated the pull request description. The header exchange will be updated in the following pull request that will also remove the distinction between requests/responses and announcements because all of these are just messages.
